### DISCUSSION TITLE
Image | Orange Pi 3

### DISCUSSION THREAD

**Discussion by zarigata** [MichaIng/DietPi#5349] [Image Request]
#### Formal device information
- Device name | OrangePi3 
- Official product URL | http://www.orangepi.org/Orange%20Pi%203%20LTS/ 
- GitHub resource URL (if available) | 
- ...

**Comment by MichaIng** [comment]
Many thanks for your request.

We btw will add new SBCs only when generated/integrated into our build script so that we have a clean and common basis and can trigger automated image upgrades: https://github.com/MichaIng/DietPi/blob/dev/.build/images/dietpi-build

**Comment by zarigata** [comment]
cant i help with the devices i have here, in the guide it was said that if needed i can create an img after ive made the system, is there a way to make it and help?

**Reply by MichaIng** [reply]
You can use [DietPi-Imager](https://github.com/MichaIng/DietPi/blob/master/.meta/dietpi-imager) to create a minimised and compressed image file. But as said, we create official images from scratch via [DietPi-Build](https://github.com/MichaIng/DietPi/blob/master/.build/images/dietpi-build) now. So any new device would need to be added to this script, to be provided as official download.

Also we'd prefer the Armbian kernel, not the ones from Orange Pi: https://www.armbian.com/orange-pi-3/
You could try and compare the Armbian Bullseye image with the ones from Orange Pi. Adding the code to the script is actually a smaller step now. A bigger problem is the small free disk space we have on our server. I'll talk to our VPS provider 😄.

**Comment by MichaIng** [comment]
@zarigata We have Orange Pi 3 LTS images now: https://dietpi.com/downloads/images/testing/
To they work on non-LTS as well?

... They seem to have a different Ethernet chip, still worth to test.

**Reply by zarigata** [reply]
Will check brother! Give me 5 minutes

**Comment by zarigata** [comment]
after many tries the LTS works in all variations, the non LTS (the old one) seen to not have wifi on the booksworm, i do belive that both the controller and the wifi chip have something to do, 

EDIT: the network controller seen to be the same, only the wifi chip changes

**Reply by MichaIng** [reply]
Indeed, AP6256 vs AW859A. Does it work when you define the non-LTS device tree?
```sh
G_CONFIG_INJECT 'fdtfile=' 'fdtfile=allwinner/sun50i-h6-orangepi-3.dtb' /boot/dietpiEnv.txt
reboot
```

**Comment by frake65** [comment]
Hello,
I own 2 of the old OrangePi 3 (not "LTS", but equipped with 4 USB 3 ports). They are supported  by Armbian, but I'd rather like to switch them to DietPi. It seems to me, that the LTS version is not fully supporting this type of device. It starts up to the initial setup procedure but then fails with network errorrs. I could not fix them using the usual way (stop DHCP, use fixed IP5 address).
Is there any hint about this device - are they comparable to the LTS type or completely different? - Any chance, they are supported?
Thanks and best regards!

**Reply by MichaIng** [reply]
Let me see, there are dedicated U-Boot configs or both, so I guess the LTS U-Boot build picks the wrong device tree by default. Can you check this:
```sh
cat /proc/device-tree/model
```
It it says it is an LTS model, the wrong device tree is used, which may cause the network issues. Is the adapter detected at all?
```sh
ip l
```
Try to define the correct device tree explicitly, reboot and see whether network works well then:
```sh
G_CONFIG_INJECT 'fdtfile=' 'fdtfile=allwinner/sun50i-h6-orangepi-3.dts' /boot/dietpiEnv.txt
```
EDIT: Ah I see I suggested the same above a year ago. According to that, the WiFi chip changed.

**Comment by frake65** [comment]
![25-07-06 15-20-17 2058](https://github.com/user-attachments/assets/29b07f19-c58f-4a76-b11e-d0573120e356)
This is happening after adding the fdtfile line to /boot/dietpiEnv.txt

Do you have any pointer for me, how to continue from initramfs prompt?
Thanks!

**Reply by MichaIng** [reply]
To verify, `cat /proc/device-tree/model` did reveal that it booted with LTS device tree, right?

So it boots with the wrong device tree, but fails with the false one, that is surprising, especially since both differ only in Ethernet, WiFi and audio nodes, as far as I can see:
- https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts?h=linux-6.12.y
- https://github.com/armbian/build/blob/main/patch/kernel/archive/sunxi-6.12/patches.armbian/add-initial-support-for-orangepi3-lts.patch

You need to remove the `dietpiEnv.txt` line again from an external system. Something seems to be initialized falsely by the bootloader. The `4022000.mmc` device, which you see in the screenshot, is the eMMC, which seems to have been reset and not probed again. Now I see the LTS device tree enables HS200 mode at 1.8V. Maybe the bootloader initiates the eMMC like that as well, and things fail if the Linux device tree of the non-LTS version does not. Let me create a non-LTS image, seems to be simple enough to add support.

**Reply by MichaIng** [reply]
Please test this image: https://dietpi.com/downloads/images/testing/

**Comment by frake65** [comment]
Hi MichaIng,
thank you so much for this sponteneous and great job. Up to now no problems occured on my site - I'm really impressed!
Is there any way to spend you some beer / coffee?
Are you planning to have that device as a newly supported one for future releases?
Have a good week!
Best regards 🥇

**Reply by MichaIng** [reply]
Yeah, I will add it to the download page with upcoming DietPi release.

> Is there any way to spend you some beer / coffee?

https://dietpi.com/contribute.html

It was easy because the Armbian build system supports it. Credits hence go to the respective Armbian contributors: https://github.com/armbian/build/blame/main/config/boards/orangepi3.csc
